### PR TITLE
[HOLD] Discovery report errors when extra files on disk.

### LINKED ISSUE #1233: Discovery report should show error when more files are staged than listed in the file manifest
Preassembly has an error condition for jobs with a file manifest we describe as:

>files_found_mismatch
>
>The number of files in the manifest does not match the number of files found in the object folder.
([see our list of possible errors](https://github.com/sul-dlss/pre-assembly/wiki/Explanation-of-possible-errors-in-a-discovery-report#files_found_mismatch))

As described, the error should occur whenever the number of files stage on the staging filesystem does not match the number of files listed in the file manifest.

In testing, I found that:

- Preassembly would detect when files on the manifest were not staged on the filesystem
- Preassembly would not detect when files staged on the filesystem were not listed on the manifest

Additional testing revealed two potential complications:

- MD5 sidecar files that should not be part of the cocina
- "hidden" files, which seem to be dotfiles (not the Windows definition of hidden files, AFAIK)

The discovery report should ignore these files if they are present but not listed on the manifest.

**Expected behavior**

The discovery report should report a files_found_mismatch when files are staged that are not on the file_manifest.csv UNLESS

- the files are sidecar MD5 files (there should be an indication of how these are defined for checksum checking purposes in assemblyWF)
- the files are hidden files

If someone does list an MD5 file on the file_manifest.csv, that should not trigger an error as long as the file is either staged or already present in a previous version of the object.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1233

# Why was this change made? 🤔

See ticket

# How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (e.g. changes what is written to shared file systems or how it uses APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit, Andrew

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

**Comment by edsu** [comment]
Are we waiting on getting some testing time on this one prior to merging?

File: app/services/object_file_validator.rb [hunk 1]
```diff
@@ -27,15 +27,11 @@ def validate
...
-      batch_id = object.druid.id
-      if batch_id && file_manifest.manifest[batch_id]
-        manifest_files = file_manifest.manifest[batch_id].fetch(:file_sets, []).flat_map { |_seq, val| val[:files] }
-        counts[:files_in_manifest] = manifest_files.count
-        relative_paths = object.object_files.map(&:relative_path) # Files found on disk
-        all_paths = (cocina_filenames + relative_paths).uniq
-        counts[:files_found] = (manifest_files.pluck(:filename) & relative_paths).count
+      if object.druid.id && file_manifest.manifest[object.druid.id]
+        counts[:files_in_manifest] = manifest_filenames.count
+        counts[:files_found] = (manifest_filenames & on_disk_filenames).count
```
Comment (edsu) [inline app/services/object_file_validator.rb:32]: Is the `&` supposed to be a `+` here?

**Review by edsu — CHANGES_REQUESTED** [review]

File: app/services/object_file_validator.rb (hunk 1 above)
Comment (justinlittman) [reply app/services/object_file_validator.rb:32]: While I admit that "files found" is completely ambiguous, the counting logic was not actually changed in this PR, just the names of the variables. Files found is the intersection of the files in the manifest and files on disk.

**Review by edsu — APPROVED** [review]

**Comment by andrewjbtw** [comment]
I tested multiple cases: hidden files, md5 sidecar, extra "regular" files and it looks good to me.

**Pull request merged**